### PR TITLE
Project Completed

### PR DESCRIPTION
**Pull request by DannieMo**
### Milestone 1 Header and Footer
Task Completed
Added header
Added Footer

### Milestone 2: HomePage
Task Completed
When the page loads, the webapp retrieves data from the selected API and shows the list of items on screen with the basic data
I Prepared the "Comments" and "Reservations" buttons. for now, they are doing nothing - just being displayed.

### MileStone 3: Item Count
In this PR I did the following:
1. I created a function to check action count
2. I imported and rendered it. check image below.
image

### MileStone 4: Added getLikes and postLike functions
Added getLikes and postLike functions.
In this milestone, I have done the following:
1. Added a function to post like to the involvement API.
2. Added a function to get likes from the involvement API.
3. Rendered the gotten likes to the page for the user.

### Final MileStone: Add Test for Items Counter
Task Completed
1. Added a function for items counter covered with unit tests using jest.
2. Added test for all edge cases.

PS. I had some issue with my coding partner. Therefore, I was authorized by Marco Lizza to build the project alone: the part that was assigned to me, which is the Student A Cards.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+// eslint-disable-next-line no-unused-vars
+const homeContainer = document.querySelector('.homepage');
+const itemsCountSpan = document.querySelector('.recipe-count');
+const baseUrl = 'https://www.themealdb.com/api/json/v1/1/search.php?f=b';
+
+const displayMeals = (meals) => {
+  const borderDiv = document.createElement('div');
+  borderDiv.className = 'mainCont';
+  // borderDiv.style.backgroundColor = 'red';
+  // eslint-disable-next-line array-callback-return
```
Comment (mikethreels) [inline src/index.js:14]: - [ ] Please do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.
- [ ] please don't disable linters we use them to make our code more efficient and easier to read.

File: src/style.css [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  border-radius: 30px;
+  overflow: hidden;
+  flex-wrap: wrap;
+}
+
+.img {
+  width: 200px;
+  height: 300px;
+  object-fit: contain;
+}
```
Comment (mikethreels) [inline src/style.css:150]: - [ ] the images "fall" off the page when opening the application as you can see in the below screenshot. please fix this.
afbeelding

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Dannie,

## Please include the comment regarding you working alone when you request for the next review as well. 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by DannieMo** [comment]
As requested by the previous reviewer, attached is a copy of the communication between Student Success, Marco, and myself and his suggestion to work on the Student A cards of the Kanban board, which is what I have done.

image

Thanks for the review.

File: src/style.css [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  .mainCont {
+    display: grid;
+    grid-template-columns: repeat(3, 1fr);
+    gap: 30px;
+    grid-auto-rows: minmax(100px, auto);
+    background-color: #edeff5;
+    padding: 20px 15%;
+  }
```
Comment (jerryowusu) [inline src/style.css:125]: As requested by the previous reviewer kindly fix the issue of some contents and images falling off and overflowing the to the outside of the main content. As shown below you can see that the third cards in the grid columns are outside the `main content` container. Kindy fix this to make your project more responsive and up to standard. :wink:

File: README.md [hunk 4]
```diff
@@ -1 +1,171 @@
...
+
+- **[Hit_the_Api]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://github.com/SulimanJoya/JavaScript-capstone-project.git)
```
Comment (jerryowusu) [inline README.md:46]: [OPTIONAL] Your live demo link is not working. Kindly make sure your app is hosted and link updated to make your project look more professional and up to standard. :+1:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @DannieMo , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

almost there2


 ## Highlights
Great work so far :heavy_check_mark: 
No linter errors :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/style.css (hunk 3 above)
Comment (DannieMo) [reply src/style.css:125]: @jerryowusu I cannot understand what you are requesting.

image

The above is the card and they appear correctly from my end.

Comment (DannieMo) [reply src/style.css:125]: Kindly check the dev branch and ensure you have my latest changes.

Comment (jerryowusu) [reply src/style.css:125]: @DannieMo, yes I understand your concerns. However, since you are building a responsive website, all screen widths of the project should look identical to this, and I can say the desktop version looks perfectly fine just as you have posted above but the mobile version is where the problem is. Kindly check the images of the mobile screens below: 


Screenshot from 2023-02-18 10-42-02


Screenshot from 2023-02-18 10-42-36

Comment (DannieMo) [reply src/style.css:125]: @jerryowusu the screen size with the width 501 is not a real device width, I tested my code against all mobile screens (320-428px), tablet screen (768px) and desktops of all size. So, I don't understand why you've used 501 and 565 to test the code.

**Comment by ahmetbozaci** [comment]
### Hello @
### _This is an additional review as a result of the second opinion request._ 

-  You already fixed the layout in the desktop version. (mentioned in this [request](the images "fall" off the page when opening the application as you can see in the below screenshot. please fix this.))
image

- Making a responsive website will be better for user experience, however, it is not mentioned by the first reviewer and it is not part of the requirement. It should be an OPTIONAL like mentioned in the [requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md#:~:text=Make%20sure%20that%20you%20have%20a%20decent%20mobile%20design%20for%20the%20webapp.).
image





___
_**Please ask for another review in your Dashboard and the next Code Reviewer will mark it as approved right away**_




### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: src/index.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        </ul>
+      </nav>
+    </header>
+    <!------Main------>
+    <section class="homepage"></section>
+    <!------Footer------>
+    <footer class="footer">
+      A Microverse Project, Created by
+      <a href="https://github.com/DannieMo">Daniel Owiri</a> under CC License
+    </footer>
```
Comment (EricMbouwe) [inline src/index.html:43]: - [OPTIONAL] I understand that you have implemented responsiveness for standard mobile sizes but you can not take it for granted that all your users will have those devices' sizes thus you should at least have a fallback(reasonable layout for sizes that are not explicitly handled in your code). 

As a starting point, you can simply add a container to your page so that your content doesn't overflow the viewport.

Here is a simple example that you can use in your header, sections and footer. It will definitely solve that issue.

In your CSS:
```css
.container {
  max-width: 1000px;
  margin: 0 auto;
  padding: 0 15%;
}
```

In your HTML:
```html
 <body>
    <!-- Header  -->
    <header>
      <div class="container">
        <nav class="navbar d-flex">
          <!-- <a href="#"> -->
            <img
              class="logo"
              src="https://raw.githubusercontent.com/SulimanJoya/JavaScript-capstone-project/homepage/src/assets/logo.png"
              alt="logo"
            />
            <!-- </a> -->
            <ul class="navlinks d-flex">
              <li class="navitems"><span class="recipe-count"></span></li>
              <li class="navitems"><a href="#Home">Home</a></li>
              <li class="navitems"><a href="#Home">Contact Us</a></li>
              <li class="navitems"><a href="#Home">About</a></li>
            </ul>
        </nav>
      </div>
    </header>
    <!------Main------>
    <section class="homepage">
      <div class="container">
         <!-- Inject your content here -->
      </div>
    </section>
    <!------Footer------>
    <footer class="footer">
      <div class="container">
          A Microverse Project, Created by
          <a href="https://github.com/DannieMo">Daniel Owiri</a> under CC License
     </div>
    </footer>
```

**Review by EricMbouwe — APPROVED** [review]
Hi @DannieMo, 

In my opinion, your project is ready for presentation! ⚡🔥 There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
